src/load_inputs/load_demand_data.jl

```diff
@@ -1,9 +1,28 @@
+function get_demand_dataframe(path)
+    filename = "Demand_data.csv"
+    deprecated_synonym = "Load_data.csv"
+    df = load_dataframe(path, [filename, deprecated_synonym])
+    # update column names
+    old_columns = find_matrix_columns_in_dataframe(df, DEMAND_COLUMN_PREFIX_DEPRECATED[1:end-1],
+                                     prefixseparator='z')
+    old_column_symbols = Symbol.(DEMAND_COLUMN_PREFIX_DEPRECATED*string(i) for i in old_columns)
+    if length(old_column_symbols) > 0
+        @info "$DEMAND_COLUMN_PREFIX_DEPRECATED is deprecated. Use $DEMAND_COLUMN_PREFIX."
+        new_column_symbols = Symbol.(DEMAND_COLUMN_PREFIX*string(i) for i in old_columns)
+        rename!(df, Dict(old_column_symbols .=> new_column_symbols))
+    end
+    return df
+end
```

Comment:
This PR would make Load_data.csv the first deprecated filename (that I know of). It therefore requires special handing. This function handles two different aspects of being backward-compatible.
1. It loads the file (preferring the updated one) and admonishes the user if they provide an old-named file.
2. It renames deprecated `Load_MW_z*` column names to `Demand_MW_z*`.

... This should probably therefore be split into two functions.